addons/sys_rack/fnc_enterVehicle.sqf

```diff
@@ -0,0 +1,62 @@
+/*
+ * Author: ACRE2Team
+ * Handles entering a vehicle.
+ *
+ * Arguments:
+ * 0: Vehicle <OBJECT>
+ * 1: Unit entering a vehicle <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [cursorTarget, player] call acre_sys_rack_fnc_enterVehicle
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_vehicle", "_unit"];
+
+if (_unit != _vehicle) then {
+    private _initialized = _vehicle getVariable [QGVAR(initialized), false];
+
+    if (!_initialized) then {
+        //Only initialize if we are first in the crew array - This helps prevent multiple requests if multiple players enter a vehicle around the same time.
+        private _crew = crew _vehicle;
+        private _firstPlayer = objNull;
+        {
+            if (!isNull _firstPlayer) exitWith {};
+            if (isPlayer _x) exitWith {
+                _firstPlayer = _x;
+            };
+        } forEach _crew;
+
+        if (!isNull _firstPlayer) then {
+            if (_unit == _firstPlayer) then {
+                [_vehicle] spawn {
+                    params ["_vehicle"];
+
+                    [_vehicle] call FUNC(initVehicle);
+                    // Give some time for the racks to initialise properly
+                    sleep 0.5;
```

Comment:
There is no need to use scheduled here.
```sqf
[{
    [_this] call FUNC(configureRackIntercom);
}, _vehicle, 0.5] call CBA_fnc_waitAndExecute;
```

Or even better, `CBA_fnc_waitUntilAndExecute` that checks if racks are initialized.

Reply:
That second part is not really possible. Since i would have to check when each of the vehicle racks are properly initialised and radioId returned from the server. This is why I put that solution. So I will take the waitAndExecute option

Reply:
How sure is this then? If very low FPS and high desync, will it happen at all during those 0.5s?